### ISSUE TITLE
feat: Vue 3 Support

### ISSUE THREAD

**Issue by crutch12** [vue-styleguidist/vue-styleguidist#997]
To compile vue3 template we have to use [@vue/compiler-sfc](https://www.npmjs.com/package/@vue/compiler-sfc) (because [vue-template-compiler](https://www.npmjs.com/package/vue-template-compiler) is for vue2 only).

https://github.com/vuejs/vue-next/issues/670#issuecomment-579785869
> I think this is an issue that should be fixed in `vue-cli-plugin-vue-next` by not including `vue-template-compiler` (which is for 2.x only)

---

So we have to replace `parseComponent(...)` from `vue-template-compiler` with `parse(...).descriptor` from `@vue/compiler-sfc`

Here:
1) https://github.com/vue-styleguidist/vue-styleguidist/blob/69c848bfcd9bcf5ee0e33053324d5ed206c85eb6/packages/vue-styleguidist/src/loaders/utils/parseVue.ts#L14
2) https://github.com/vue-styleguidist/vue-styleguidist/blob/daf043224af2a1b8bde19af5fe4426dab466596b/packages/vue-styleguidist/src/loaders/utils/getScript.ts#L7
3) https://github.com/vue-styleguidist/vue-styleguidist/blob/f5b70bf1bdca13495fd041a3eb6e844bf9f9e4c4/packages/vue-styleguidist/src/loaders/utils/processComponent.ts#L44

**Comment by elevatebart** [comment]
You are absolutely right, thanks for the recommendation.

It might not be enough. I will track the vue 3 compatibility work on this issue.

**Comment by elevatebart** [comment]
As expected, the rewiring of the compiler was not enough. 
Since Vue apps have a very different root in v2 and v3, this file needs to be re-wired.
https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/packages/vue-styleguidist/src/client/utils/globalizeComponent.ts

This is how we do it in vue 3.
https://dev.to/jakedohm_34/registering-global-components-with-vue-3-2kgl

It would mean having one vue app per example block. 
In each vue app, all components would be registered again.

**Comment by elevatebart** [comment]
For the reasons above vue 3 compatibility means breaking changes in the way we use `renderRootJsx`.
Breaking changes mean a major version.

I suggest we use the "next" branch and the "@next" channel to deliver vue 3 compat.

**Comment by elevatebart** [comment]
I hit a snag last time I tried which is I have to abandon TypeScript completely if I am to adopt vue 3.
Indeed, the TSX typings of vue 3 conflict with React's typings so one cannot have them both installed at the same time.

**Comment by crutch12** [comment]
Classical TypeScript :/

![image](https://user-images.githubusercontent.com/19373212/96041510-9b558380-0e74-11eb-9e90-6fa95c4b5d72.png)

**Comment by elevatebart** [comment]
To be fair to TypeScript, that is classic Facebook/React. 
TSX was made with the Facebook motto in mind: "Go fast and break things" pushing them to add the TSX typings as ambient, not caring about anyone that would need another version of these typings (like Vue or Svelte)

**Comment by crutch12** [comment]
What can we do now?

**Comment by elevatebart** [comment]
Hello @crutch12 

I removed all TypeScript from the vue3 branch. I see your point.
Then I have to remove all tests because `@testing-library/react` imports react types which makes babel fail.

I was OK with no TypeScript until it is fixed but no tests at all, that's a bummer.
I have an issue posted on vue-next and a PR that goes with it. 

Now, we wait.

**Comment by spaceemotion** [comment]
@elevatebart Are we still waiting on https://github.com/vuejs/vue-next/issues/1033? You said you would have some time on VueLand, did you manage to have an idea on how to approach this?

**Comment by elevatebart** [comment]
Hello @spaceemotion 

I found a solution that works and have not delivered it.

1. going to vue3 is a deep breaking change, so I would like to use this opportunity to add breaking changes that have been asked for a long time. You can find a list of these changes in https://github.com/vue-styleguidist/vue-styleguidist/pull/1002
1. The idea of removing all unit tests and all types from vue-styleguidist is not acceptable to me. So, for now, we are still waiting.

If you are documenting your components in the style of vue-styleguidist, you still have options:
- `vue-docgen-api` is already compatible with vue3. It does not detect everything but it works. I am very eager to get feedback with it.
- Use `vue-docgen-cli` generates markdown files that can be consumed by any static site generator.
- Complete it with `vue-live@next` for live examples and you can build your own vue 3 compatible tool

I hope this helps.

**Comment by spaceemotion** [comment]
Thanks for the tip! I only wanted to use vue-docgen-api anyways so that was perfect timing.

I incorporated it into a custom rollup plugin and it worked without any problems 🎉 (.vue files lang=typescript).

Will keep you posted while I explore the options/annotations 👍

**Comment by spaceemotion** [comment]
Alright, not sure if I should open up a new issue for this, but I just tried the following and it doesn't pick up the displayName or description (props work fine though):

```ts
// (do something up here with "typeof component")

/**
 * Buttons allow users to perform an action or to navigate to another page.
 * They have multiple styles for various needs, and are ideal for calling attention
 * to where a user needs to do something in order to move forward in a flow.
 *
 * @displayName Button
 * @group Form Elements
 */
const component = defineComponent({
  name: "DsButton",
  props: {},
});

export default component;
```

Gives me:
```json
{
  "displayName": "DsButton",
  "exportName": "default",
  "description": "",
  "tags": {},
  "props": [],
  "slots": [
    {
      "name": "default"
    }
  ]
}
```

**Comment by elevatebart** [comment]
There is an issue common to vue 2 if you assign the output instead of exporting it directly...
Let me fix this really quick..

In the meantime, this should work though.

```ts
/**
 * Buttons allow users to perform an action or to navigate to another page.
 * They have multiple styles for various needs, and are ideal for calling attention
 * to where a user needs to do something in order to move forward in a flow.
 *
 * @displayName Button
 * @group Form Elements
 */
export default defineComponent({
  name: "DsButton",
  props: {},
});
```

**Comment by Keyes** [comment]
Any news on Vue3 support so far? I enjoy using styleguidist way more than storybook for example, but it feels weird starting a new project now with Vue2 only because of that... :/

**Comment by elevatebart** [comment]
@Keyes thanks for the kind words.

I am a bit swamped right now but it's moving. checkout the progress on the Pull Requests marked with the label next major

My plan is to support Vue3 in the next major version. I should be starting to work on it in March, for a release at the beginning of summer.

- No more support for Vue 2
- Use of typescript in examples
- Monaco as the editor with an option to get back to the prism editor if you need lighter sites.
- And hopefully sass compilation in SFC

**Comment by mambari** [comment]
Hi @elevatebart
Thank you for your work.
Any news on Vue3 support (even beta that we can test) ?
Best regards,

**Comment by emersonbottero** [comment]
> Hello @spaceemotion
> 
> I found a solution that works and have not delivered it.
> 
> 1. going to vue3 is a deep breaking change, so I would like to use this opportunity to add breaking changes that have been asked for a long time. You can find a list of these changes in #1002
> 2. The idea of removing all unit tests and all types from vue-styleguidist is not acceptable to me. So, for now, we are still waiting.
> 
> If you are documenting your components in the style of vue-styleguidist, you still have options:
> 
> * `vue-docgen-api` is already compatible with vue3. It does not detect everything but it works. I am very eager to get feedback with it.
> * Use `vue-docgen-cli` generates markdown files that can be consumed by any static site generator.
> * Complete it with `vue-live@next` for live examples and you can build your own vue 3 compatible tool
> 
> I hope this helps.

Has someone made a tool like that, it is exactly what I need. 😁
maybe @spaceemotion ? 🤔

**Comment by emersonbottero** [comment]
> Hello @spaceemotion
> 
> I found a solution that works and have not delivered it.
> 
> 1. going to vue3 is a deep breaking change, so I would like to use this opportunity to add breaking changes that have been asked for a long time. You can find a list of these changes in #1002
> 2. The idea of removing all unit tests and all types from vue-styleguidist is not acceptable to me. So, for now, we are still waiting.
> 
> If you are documenting your components in the style of vue-styleguidist, you still have options:
> 
> * `vue-docgen-api` is already compatible with vue3. It does not detect everything but it works. I am very eager to get feedback with it.
> * Use `vue-docgen-cli` generates markdown files that can be consumed by any static site generator.
> * Complete it with `vue-live@next` for live examples and you can build your own vue 3 compatible tool
> 
> I hope this helps.

Has someone made a tool like that, it is exactly what I need. 😁
maybe @spaceemotion ? 🤔

**Comment by elevatebart** [comment]
This is using nuxt content: https://github.com/elevatebart/vue-styleguidist-example-docgen-nuxt
I need to cook an example with the same principle using vitepress.

Give me a day ;)

**Comment by spaceemotion** [comment]
We have a next.js page running at https://design.demv.de. 

A custom rollup plugin extracts the component JSON the docgen produces. Next.Js just reads from it. 

The only problem we have is speed. It takes up to 6 seconds for examples to update…

**Comment by emerxom** [comment]
> This is using nuxt content: https://github.com/elevatebart/vue-styleguidist-example-docgen-nuxt
> I need to cook an example with the same principle using vitepress.
> 
> Give me a day ;)

Thank you so much for the example... It's working.... 😁
some thoughts..
`options: { type: Array as () => string[] },` Works
`options: Array as () => string[],` Don't work, on those cases it does not create the md file.
This happens exactly the same when using PropTypes<>

It always throws an error using vue-docgen from "vue-docgen-cli": "^4.40.0",
but still creates the .md files
```
yarn run v1.22.10
$ C:\...\node_modules\.bin\vue-docgen .\src\components\file.vue
(node:31160) UnhandledPromiseRejectionWarning: Error: Error compiling file C:\...\file.vue:Cannot read property 'length' of undefined
    at C:\...\node_modules\vue-docgen-cli\lib\multiMd.js:139:27
    at step (C:\...\node_modules\vue-docgen-cli\lib\multiMd.js:52:23)
    at Object.throw (C:...\node_modules\vue-docgen-cli\lib\multiMd.js:33:53)
    at rejected (C:\...\node_modules\vue-docgen-cli\lib\multiMd.js:25:65)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:31160) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:31160) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Terminate batch job (Y/N)? y
```
basic component layout
``` vue
<template>
</template>

<style scoped>
</style>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
});
</script>
```

**Comment by stelco** [comment]
Hi, any update on this for Vue 3 support out of the box? Thanks.

**Comment by dvh91** [comment]
Has anyone managed to generate docgen component metadata from Vue 3 sfc with Typescript?

```vue
<script lang="ts" setup>
    interface Stack {
        typescript: string
        vue: string
        docgen: string
    }

    interface Props {
        stack: Stack
    }

    defineProps<Props>()
</script>
```

**Comment by PauloSSAragao** [comment]
Dears, there is any specific date planning to have vue-styleguidis working on vue-3.
Not sure if I wait for it or try another solution (I hope not, as I have liked so much the styleguidist solution for react)

**Comment by elevatebart** [comment]
@dvh91 @PauloSSAragao Thank you for your interest in the tool, it means a lot.

I have not worked seriously on styleguidist for a while. My work at Cypress and the raising of my little human have taken the better part of my brain power this year.

@PauloSSAragao I would not expect any good news before spring unless you are willing to help.

To jump on version 3 of Vue we need to change the architecture of the tool.
Here is a summary of why and how we will do that.

## How it worked

The architecture of vue-styleguidist@4 works very well with the explicitness of Vue 2. 

### vue-docgen-api

There are only 3 places I can find a prop for a component. 

- It is in its code, in the props member, or in the `@props` if it's a class style component
- It's in its parent components code (using extends)
- It's in one of the mixins it uses

We would parse the current components code using `@babel/parser` and if by any chance the component had a `mixin` or a `extend` we would parse the imported components corresponding to these `extends` and `mixins`.

Since there is only 3 places to look in, it was rather easy find where the current prop has been documented. With this data found, composing the documentation object was a matter of merging them in the right order.

### vue-styleguidist

Most of the rendering code was already made in `react-styleguidist`. It made the dev and maintenance of the front-end of the tool relatively simple. 

Vue-styleguidist would leverage the vue-docgen-api to build the documentation website.

## What failed with Vue 3

### vue-docgen-api

Now that TypeScript is a first class citizen, we can define prop types in interfaces. Those interfaces can be defined inline like here. That makes parsing the same as with Vue 2.

```vue
<script lang="ts" setup>
    defineProps<{
	  /**
	   * typescript is easy to document
	   */
        typescript: string
        vue: string
        docgen: string
    }>()
</script>
```

Often, we will not want to be that explicit 
The case below becomes a bit more complicated:
- Resolving the interface by matching it's name with it's definition (a type, an interface)
- Creating Prop documentation that is not a flat array but a set of deep objects. We need a much more robust type than what we have today to return that

```vue
<script lang="ts" setup>
    interface Stack {
	  /**
	   * typescript is easy to document
	   */
        typescript: string
        vue: string
        docgen: string
    }

    interface Props {
	  /**
	   * stack? what is that
	   */
        stack: Stack
    }

    defineProps<Props>()
</script>
```

And this could become extensively more complex when one adds spreads and imports to the definition of interfaces.

```vue
<script lang="ts" setup>
    import type { ExternalProps } from "../common/types"
    interface Stack {
	  /**
	   * typescript is easy to document
	   */
        typescript: string
        vue: string
        ...ExternalProps
        docgen: string
    }

    interface Props {
	  /**
	   * stack? what is that
	   */
        stack: Stack
    }

    defineProps<Props>()
</script>
```

### vue-styleguidist

The front end of `react-styleguidist` written in React TSX. 
React JSX is not compatible with Vue3 typings. Vue 3 imposes that you DO NOT already use TSX in any framework besides Vue's. 
You'd ask me:
"Why did you not try JSX instead of TSX to avoid conflicts" 
I did. And it worked partially. As soon as you import `@testing-library/react` the ambient React JSX types are imported again and we are back to square 1. I decided that working without TS and without test was too dangerous so I stopped that way.

## The ambition

### vue-docgen-api

A couple of new features are necessary
- Implement setup syntax parsing
- Resolution of interfaces, in the same file, resolution of variables
- Definition of a storage format for complex types
- Management of spread operator

I wonder if using the native typescript parser would be more efficient here. If we were to change it means **rewriting** the new library with the new AST vocable.

A better management of the `@example` tag would be likely too.

### vue-styleguidist

First, I need to design and implement the visual aspect of complex interfaces as rendered above.

Second, Since we cannot use react anymore, why not use Vue all the way. And, while we are at it, why not ditch webpack altogether in favor of vite. I need to re-implement the entire front-end which is fun but takes time.

**Comment by PauloSSAragao** [comment]
> 
> 
> @dvh91 @PauloSSAragao Thank you for your interest in the tool, it means a lot.
> 
> I have not worked seriously on styleguidist for a while. My work at Cypress and the raising of my little human have taken the better part of my brain power this year.
> 
> @PauloSSAragao I would not expect any good news before spring unless you are willing to help.
> 
> To jump on version 3 of Vue we need to change the architecture of the tool. Here is a summary of why and how we will do that.
> ## How it worked
> 
> The architecture of vue-styleguidist@4 works very well with the explicitness of Vue 2.
> ### vue-docgen-api
> 
> There are only 3 places I can find a prop for a component.
> 
>     * It is in its code, in the props member, or in the `@props` if it's a class style component
> 
>     * It's in its parent components code (using extends)
> 
>     * It's in one of the mixins it uses
> 
> 
> We would parse the current components code using `@babel/parser` and if by any chance the component had a `mixin` or a `extend` we would parse the imported components corresponding to these `extends` and `mixins`.
> 
> Since there is only 3 places to look in, it was rather easy find where the current prop has been documented. With this data found, composing the documentation object was a matter of merging them in the right order.
> ### vue-styleguidist
> 
> Most of the rendering code was already made in `react-styleguidist`. It made the dev and maintenance of the front-end of the tool relatively simple.
> 
> Vue-styleguidist would leverage the vue-docgen-api to build the documentation website.
> ## What failed with Vue 3
> ### vue-docgen-api
> 
> Now that TypeScript is a first class citizen, we can define prop types in interfaces. Those interfaces can be defined inline like here. That makes parsing the same as with Vue 2.
> 
> ```
> <script lang="ts" setup>
>     defineProps<{
> 	  /**
> 	   * typescript is easy to document
> 	   */
>         typescript: string
>         vue: string
>         docgen: string
>     }>()
> </script>
> ```
> 
> Often, we will not want to be that explicit The case below becomes a bit more complicated:
> 
>     * Resolving the interface by matching it's name with it's definition (a type, an interface)
> 
>     * Creating Prop documentation that is not a flat array but a set of deep objects. We need a much more robust type than what we have today to return that
> 
> 
> ```
> <script lang="ts" setup>
>     interface Stack {
> 	  /**
> 	   * typescript is easy to document
> 	   */
>         typescript: string
>         vue: string
>         docgen: string
>     }
> 
>     interface Props {
> 	  /**
> 	   * stack? what is that
> 	   */
>         stack: Stack
>     }
> 
>     defineProps<Props>()
> </script>
> ```
> 
> And this could become extensively more complex when one adds spreads and imports to the definition of interfaces.
> 
> ```
> <script lang="ts" setup>
>     import type { ExternalProps } from "../common/types"
>     interface Stack {
> 	  /**
> 	   * typescript is easy to document
> 	   */
>         typescript: string
>         vue: string
>         ...ExternalProps
>         docgen: string
>     }
> 
>     interface Props {
> 	  /**
> 	   * stack? what is that
> 	   */
>         stack: Stack
>     }
> 
>     defineProps<Props>()
> </script>
> ```
> 
> ### vue-styleguidist
> 
> The front end of `react-styleguidist` written in React TSX. React JSX is not compatible with Vue3 typings. Vue 3 imposes that you DO NOT already use TSX in any framework besides Vue's. You'd ask me: "Why did you not try JSX instead of TSX to avoid conflicts" I did. And it worked partially. As soon as you import `@testing-library/react` the ambient React JSX types are imported again and we are back to square 1. I decided that working without TS and without test was too dangerous so I stopped that way.
> ## The ambition
> ### vue-docgen-api
> 
> A couple of new features are necessary
> 
>     * Implement setup syntax parsing
> 
>     * Resolution of interfaces, in the same file, resolution of variables
> 
>     * Definition of a storage format for complex types
> 
>     * Management of spread operator
> 
> 
> I wonder if using the native typescript parser would be more efficient here. If we were to change it means **rewriting** the new library with the new AST vocable.
> 
> A better management of the `@example` tag would be likely too.
> ### vue-styleguidist
> 
> First, I need to design and implement the visual aspect of complex interfaces as rendered above.
> 
> Second, Since we cannot use react anymore, why not use Vue all the way. And, while we are at it, why not ditch webpack altogether in favor of vite. I need to re-implement the entire front-end which is fun but takes time.

hello @elevatebart 
thanks for your response!!
Sorry if my question was like "provide me a solution because I need it ..."  :-)
Not my intention because I suppose how hard should be to keep a amazing open source project with job, family, no financial support, etc ..
But man, I really appreciate ur support and commitment to provide us a solution even without any guarantee of any return!!!
Many thanks!!!

**Comment by jtuchel** [comment]
First of all thanks a lot for this nice package :) Great work!

Do you have any plans / a roadmap when we can use it with Vue3 ( e.g. Summer 2022) ?

Or should we just keep an eye on this? https://github.com/vue-styleguidist/vue-styleguidist/pull/1002

**Comment by santiagoarizti** [comment]
Hi, first commenter,

> And this could become extensively more complex when one adds spreads and imports to the definition of interfaces

I just wanted to point out that this is not strictly necessary for a first delivery of typescript support, since the official documentation of Vue says that this feature (using typescript-only `defineProps`) only works well with **native types** and **locally defined interfaces**. They do mention that in theory they could in the future support imported interfaces, but they don't as of the time of writing.

Source: <https://v3.vuejs.org/api/sfc-script-setup.html#type-only-props-emit-declarations>

> As of now, the type declaration argument must be one of the following to ensure correct static analysis:
> 
> - A type literal
> - A reference to an interface or a type literal in the same file

**Comment by hoopoooo** [comment]
Hey guys,
this question might be obvious and so I didn't want to open a new issue.

Am I correct that I can't use this package for Vue3 apps yet? ( same as here https://github.com/vue-styleguidist/vue-styleguidist/issues/997#issuecomment-998619393 )

I'm using the Vue CLI v4.5.15, created a new app, run `vue add styleguidist`, run `npm run styleguide` and get the error

> Cannot find module 'vue-template-compiler'

so I think the answer is still obvious :(

But since this issue is from october 2020, are there any chances Vue3 support will arrive in some months? :)

**Comment by santiagoarizti** [comment]
@hoopoooo I use storybook vor vue, version 6.4, and it internally uses vue-styleguidist.
I use vue 3 and composition api.
with my component files that use normal `export default defineComponent({props: ..., setup() ...` the props are extracted ok.
The issue currently lies in components with the new `<script setup>defineProps<{...` syntax.

**Comment by elevatebart** [comment]
@santiagoarizti @hoopoooo 4.44.1 now has support for `<script setup>` syntax !!

https://vue-styleguidist.github.io/docs/Documenting.html#setup-syntax

**Comment by dvh91** [comment]
@elevatebart that's amazing! thanks so much!
Just a quick question- does it also work with `withDefaults`?

**Comment by elevatebart** [comment]
It is supposed to work this way. Please tell me if it does not, it would be a bug.

**Comment by dvh91** [comment]
@elevatebart Thanks! withDefaults indeed working great! Can you maybe assist with [this one](https://github.com/vue-styleguidist/vue-styleguidist/issues/1261)?

**Comment by shadow7412** [comment]
Should this still work with Vue 2? Or is there a specific version I should target for projects that haven't been updated yet?

**Comment by elevatebart** [comment]
@shadow7412 It is unit-tested with vue2 (still) so everything should work with vue 2 and not vue 3.

Do you have any issue with a vue 2 project ?

**Comment by shadow7412** [comment]
I was getting an error that had something to do with vue sfcs. From memory it might have been unable to find @vue/compiler-sfcs... But I won't be able to get the exact errors until I get back to work on Tuesday.

**Comment by shadow7412** [comment]
@elevatebart sorry about the delay.

My current error when attempting to build for a Vue 2 project is `Error: Cannot find module 'vue/compiler-sfc'`. Attempting to run `npm install vue/compiler-sfc` directs me to log into github, then fails (I assume based on this it's a private package?).

My vue is at 2.6.14, and vue-styleguidist is at 4.44.18 in case that helps.

**Comment by bodograumann** [comment]
@shadow7412 It should be `@vue/compiler-sfc`. Notice the `@` at the front.

**Comment by shadow7412** [comment]
> @shadow7412 It should be `@vue/compiler-sfc`. Notice the `@` at the front.

In the npm command, the import, or both?

**Comment by fallsimply** [comment]
Both. The `vue/compiler-sfc` import was introduced in Vue 3

> > @shadow7412 It should be `@vue/compiler-sfc`. Notice the `@` at the front.
> 
> In the npm command, the import, or both?

**Comment by elevatebart** [comment]
And this is the package I use in docgen, to parse the sfc

**Comment by shadow7412** [comment]
Does that mean it should be listed as a dependency, assuming it was introduced in vue 3 and not backported?

**Comment by elevatebart** [comment]
It should... I hope it is...

**Comment by elevatebart** [comment]
It is

**Comment by shadow7412** [comment]
@elevatebart is it though?

<details>
  <summary>This is an excerpt from package.json in `node_modules/vue-styleguidist`</summary>

   ```
  "dependencies": {
    "@vxna/mini-html-webpack-template": "^1.0.0",
    "ast-types": "0.13.4",
    "classnames": "^2.3.1",
    "clean-webpack-plugin": "^3.0.0",
    "cli-progress": "^3.9.0",
    "clipboard-copy": "^3.2.0",
    "clsx": "^1.1.1",
    "codemirror": "^5.60.0",
    "common-dir": "^2.0.2",
    "copy-webpack-plugin": "^5.1.2",
    "core-js": "^3.9.1",
    "css-loader": "^2.1.1",
    "es6-object-assign": "^1.1.0",
    "es6-promise": "^4.2.8",
    "escodegen": "^1.14.3",
    "findup": "^0.1.5",
    "function.name-polyfill": "^1.0.6",
    "github-slugger": "^1.3.0",
    "glob": "^7.1.6",
    "glogg": "^1.0.2",
    "hash-sum": "^1.0.2",
    "is-directory": "^0.3.1",
    "javascript-stringify": "^1.6.0",
    "jss": "^10.6.0",
    "kleur": "^2.0.2",
    "leven": "^2.1.0",
    "loader-utils": "^1.4.0",
    "lodash": "^4.17.21",
    "lru-cache": "^4.1.5",
    "mini-html-webpack-plugin": "^3.1.3",
    "minimist": "^1.2.5",
    "prismjs": "^1.23.0",
    "prop-types": "^15.7.2",
    "q-i": "2.0.1",
    "qss": "^2.0.3",
    "react": "^17.0.2",
    "react-codemirror2": "^7.2.1",
    "react-dev-utils": "^12.0.0-next.47",
    "react-dom": "^17.0.2",
    "react-group": "^3.0.2",
    "react-icons": "^3.11.0",
    "react-lifecycles-compat": "^3.0.4",
    "react-simple-code-editor": "^0.11.0",
    "react-styleguidist": "^11.1.6",
    "rewrite-imports": "^2.0.3",
    "source-map": "0.6.1",
    "style-loader": "^1.3.0",
    "terser-webpack-plugin": "^2.3.8",
    "to-ast": "^1.0.0",
    "vue-docgen-api": "^4.44.22",
    "vue-inbrowser-compiler": "^4.44.22",
    "vue-inbrowser-compiler-utils": "^4.44.22",
    "webpack-dev-server": "^3.11.2",
    "webpack-merge": "^4.2.2"
  },
  "deprecated": false,
  "description": "Vue components style guide generator",
  "devDependencies": {
    "@babel/cli": "7.17.6",
    "@babel/core": "7.17.5",
    "@babel/plugin-proposal-class-properties": "7.16.7",
    "@babel/plugin-proposal-object-rest-spread": "7.17.3",
    "@babel/plugin-syntax-dynamic-import": "7.8.3",
    "@babel/plugin-transform-runtime": "7.17.0",
    "@babel/preset-env": "7.16.11",
    "@babel/preset-react": "7.16.7",
    "@babel/preset-typescript": "7.16.7",
    "@babel/runtime": "7.17.2",
    "@babel/types": "7.17.0",
    "@testing-library/react": "11.2.7",
    "@types/buble": "0.20.1",
    "@types/cli-progress": "1.8.1",
    "@types/copy-webpack-plugin": "4.4.4",
    "@types/doctrine": "0.0.5",
    "@types/escodegen": "0.0.7",
    "@types/loader-utils": "1.1.6",
    "@types/lodash": "4.14.178",
    "@types/mini-html-webpack-plugin": "2.2.2",
    "@types/prismjs": "1.16.6",
    "@types/pug": "2.0.6",
    "@types/react-dev-utils": "9.0.10",
    "@types/react-dom": "17.0.11",
    "@types/react-lifecycles-compat": "3.0.1",
    "@types/terser-webpack-plugin": "1.2.1",
    "@types/webpack": "4.41.32",
    "@types/webpack-dev-server": "3.11.6",
    "@types/webpack-merge": "4.1.5",
    "buble": "0.20.0",
    "concurrently": "4.1.2",
    "ejs": "2.7.4",
    "pug": "3.0.2",
    "strip-shebang": "1.0.2",
    "vue": "2.6.14",
    "vue-template-compiler": "2.6.14",
    "webpack": "4.46.0"
  },
```
</details>

**Comment by shadow7412** [comment]
> @shadow7412 It should be `@vue/compiler-sfc`. Notice the `@` at the front.

I tried npm installing `@vue/compiler-sfc`, but I'm getting the exact same error. Note that the import I mentioned there is from this project rather than my own, so if it's wrong and supposed to be prefixed with a `@` as you say, then I'm surprised this is working for anyone.

**Comment by elevatebart** [comment]
First a quick detail about dependencies:

Vue-styleguidist depends on vue-docgen-api that depends on @vue/compiler-sfc and @vue/compiler-core.

**Comment by elevatebart** [comment]
Second, can you please try and create a reproduction on a GitHub repo.

This is the best way I can help you.

**Comment by shadow7412** [comment]
Sorry about the delay - does [this ](https://github.com/shadow7412/vue-styleguidist-error-repro) work for you? (Though I ended up with a different error message again... :/)

**Comment by elevatebart** [comment]
@shadow7412 It does work for me, except I do not have a font-awesome pro account. 

I was able to work around that, but do not get the error you mention about `@vue/compiler-sfc`.

What OS/ version of node/version of npm are you using?

**Comment by elevatebart** [comment]
Also, when using the vue-cli, please use the cli-plugin. There is a lot of weird quirks about using vue-cli and it needs the plugin.

See [this doc](https://vue-styleguidist.github.io/VueCLI3doc.html) for more info.

**Comment by Psycarlo** [comment]
I have a vite-vue3-typescript project
When I install Vue Styleguidist
`yarn add --dev vue-styleguidist`
and then run
`vue-styleguidist server`
I get
`Error: This version of vue-styleguidist is only compatible with Vue 2.`

**Comment by StefanGhiban** [comment]
> https://vue-styleguidist.github.io/docs/Documenting.html#setup-syntax

Same project, same issue.

**Comment by StefanGhiban** [comment]
> I have a vite-vue3-typescript project When I install Vue Styleguidist `yarn add --dev vue-styleguidist` and then run `vue-styleguidist server` I get `Error: This version of vue-styleguidist is only compatible with Vue 2.`

Same project, same issue.

**Comment by Gagydzer** [comment]
Is it work with vue 3 or not?
Tried with @vue/cli 4.5.15 and it requiries vue-template-compiler

**Comment by elevatebart** [comment]
> Is it work with vue 3 or not?
> 
> Tried with @vue/cli 4.5.15 and it requiries vue-template-compiler

Not yet.

**Comment by elevatebart** [comment]
I believe I will be able to add alpha support for vue 3 around the week of July 4th.

Expect a new major version (5.0.0-alpha) to be released mid July. Then we shall see ;)

**Comment by elevatebart** [comment]
Compatibility with Vue 3 was achieved without breaking change in 4.46.1

https://github.com/vue-styleguidist/vue-styleguidist/releases/tag/vue-styleguidist%404.46.0

**Comment by elevatebart** [comment]
I will close this issue now, please open separate issues for bug reports ;)

TYTYTY
